test/mocks/application/DocumentRootInMemoryRepository.scala

```diff
@@ -172,4 +173,6 @@ class DocumentRootInMemoryRepository extends GeneralApplicationRepository {
   def removeProgressStatuses(appId: String, progressStatuses: List[ProgressStatuses.ProgressStatus]): Future[Unit] = ???
 
   def getOnlineTestApplication(appId: String): Future[Option[OnlineTestApplication]] = ???
+
+  def progressCheck(applicationId: String, progressStatus: ProgressStatus): Future[Boolean] = ???
```

Comment:
If this is only used for test purposes, perhaps we could put it on the implementation with a protected access modifier?